src/if.js

```diff
@@ -40,9 +40,34 @@ export class If {
   * @param newValue The new value
   */
   valueChanged(newValue) {
+    if (this.__queuedChanges) {
+      this.__queuedChanges.push(newValue);
+      return;
+    }
+
+    let maybePromise = this._runValueChanged(newValue);
+    if (maybePromise instanceof Promise) {
+      let queuedChanges = this.__queuedChanges = [];
+
+      let runQueuedChanges = () => {
+        if (!queuedChanges.length) {
+          this.__queuedChanges = undefined;
+          return;
+        }
+
+        let nextPromise = this._runValueChanged(queuedChanges.shift()) || Promise.resolve();
+        nextPromise.then(runQueuedChanges);
+      };
+
+      maybePromise.then(runQueuedChanges);
+    }
+  }
+
+  _runValueChanged(newValue) {
     if (!newValue) {
+      let viewOrPromise;
       if (this.view !== null && this.showing) {
-        let viewOrPromise = this.viewSlot.remove(this.view);
+        viewOrPromise = this.viewSlot.remove(this.view);
```

Comment:
I think we should address that separately. The solution might be to extend the classes that the animator understands so that we give that control to the developer rather than making assumptions. Let's see if we can fix the if and add the else first and then pick up the other conversation after that.